js/code.js

```diff
@@ -0,0 +1,489 @@
+                    if (this.checkMove(movingCards, secondDeck)) {
+                        this.moveCards(firstDeck, secondDeck, movingCards);
+                    }
+                    this.openLastCardForPlayingDeck(firstDeck);
+                    this.unselectDecks(firstDeck,secondDeck);
+                    firstDeck = null;
+                    movingCards = [];
+
+                } else if (firstDeck == secondDeck &&(cards !=undefined)){// if 2 clicks on one deck
+                    this.unselectDecks(firstDeck,secondDeck);
+                    firstDeck = null;
+                    movingCards = [];
+
+            } else if (cards !=undefined) {
+                firstDeck = secondDeck;
+                movingCards = cards;
+            }
+
+        }
+    },
+
+
+    sliceForDealDeck: function (firstDeck,secondDeck,cards) {
+        if (secondDeck instanceof DealDeck && firstDeck == null &&(cards !=undefined))//if dealdeck--> took 1
+            return cards.slice(0,1);
+        return cards;
+    },
+
+    checkVictory: function () {
+      for (let i = 0;i < 4 ;i++){
+          if (this.finishDecks[i].cards.length==0) return false
+          if (this.finishDecks[i].cards[this.finishDecks[i].cards.length-1].number != 13)return false
+      }
+  		return true;
+    },
+
+    moveKing: function (firstDeck,secondDeck,cards,movingCards){//to move king on deck with no cards
+        if (firstDeck!=null && cards == undefined){
+            if (movingCards[0].number == 13){
+                this.moveCards(firstDeck, secondDeck, movingCards);
+                this.openLastCardForPlayingDeck(firstDeck);
+                firstDeck = null;
+                secondDeck.cards.forEach((card) => card.unselect());
+            }
+        }
+    },
+
+    openLastCardForPlayingDeck: function(deck){
+        if (deck instanceof PlayingDeck)//колода ли это
+            deck.openLastCard();
+    },
+
+    checkMove: function (cards, secondDeck) {
+        let from = cards[0];
+        let to = secondDeck.cards[secondDeck.cards.length-1];
+
+        if (from.color !== to.color && (to.number - from.number == 1)) return true;
+        else return false;
+    },
+
+    unselectDecks: function (firstDeck,secondDeck) {//unselect decks after move them
+        secondDeck.cards.forEach((card) => card.unselect());
+        firstDeck.cards.forEach((card) => card.unselect());
+    },
+
+    moveCards: function (from, to, cards) {//moving cards
+            to.addCards(cards);
+            from.removeCards(cards);
+    },
+
+}
+
+Deck.prototype = {
+    createCards: function (cardKits) {
+        for (let i = 0; i < cardKits.length; i++) {
+            let card = new Card(cardKits[i]);
+
+            this.$el.appendChild(card.$el);
+            this.cards.push(card);
+        }
+    },
+
+    onClick: function(e){
+        if (e.target.className == "deck" && e.target.firstChild == null) {//check deck with no cards

```

Comment:
it's not a good practice to rely on classes because your designer may update it for example. Much better to add a `data-` attribute for example and everyone will understand that this is something important and can't be changed without confirmation.